src/System.Buffers/src/System/ByteSpan.cs

```diff
@@ -74,7 +75,8 @@ public int Length
 
         public ByteSpan Slice(int index)
         {
-            Precondition.Require(index < Length);
+            if (index >= Length)
+                return new ByteSpan(_data, 0);
```

Comment:
I think this way it is easier to use this class. I.e. when you want to iterate you can do:
while (byteSpan.Length != 0)
with that it would throw when it gets to 0. I think that was the main reason I added this changed it in the first place.